src/endpoint/scheduler.rs

```diff
@@ -370,6 +374,36 @@ impl JobHandle {
     }
 }
 
+impl Drop for JobHandle {
+    fn drop(&mut self) {
+        debug!("Cleaning up JobHandle");
```

Comment:
Nit: Might be nice/useful to include the job ID (but debugging is obviously optional).